x/marketmap/types/market.go

```diff
@@ -40,15 +53,95 @@ func (mm *MarketMap) ValidateBasic() error {
 		}
 	}
 
-	return nil
+	switch mm.AggregationType {
+	case AggregationType_INDEX_PRICE_AGGREGATION:
+		return ValidateIndexPriceAggregation(*mm)
+	default:
+		return nil
```

Comment:
should this be an "error unsupported"?

Reply:
I left it as this for now just for simplicity, we can probably add other conditions later. Otherwise we break every other test case using the market map